chromium/browser_type.go

```diff
@@ -64,27 +50,31 @@ func NewBrowserType(ctx context.Context) api.BrowserType {
 		fieldNameMapper: common.NewFieldNameMapper(),
 	}
 	rt.SetFieldNameMapper(b.fieldNameMapper)
+
 	return &b
 }
 
+// Connect attaches k6 browser to an existing browser instance.
 func (b *BrowserType) Connect(opts goja.Value) {
 	rt := k6common.GetRuntime(b.Ctx)
 	k6common.Throw(rt, errors.New("BrowserType.connect() has not been implemented yet"))
 }
 
+// ExecutablePath returns the path where the extension expects to find the browser executable.
 func (b *BrowserType) ExecutablePath() string {
 	return "chromium"
```

Comment:
The comment doesn't make sense for what this currently does, i.e. return a static string.

There are other inconsistencies as well:

- There's a [`LaunchOptions.ExecutablePath` field](https://github.com/grafana/xk6-browser/blob/cbe11c4231a92f32d61d0d8a375d6e7595722fe7/common/launch.go#L99) which is read from the options, but not used for anything... :confused:

- There's an [`Allocator.execPath` field](https://github.com/grafana/xk6-browser/blob/aeb5fb2/chromium/allocator.go#L29) which is the real executable path we determine at runtime and [what we actually run](https://github.com/grafana/xk6-browser/blob/aeb5fb2/chromium/allocator.go#L29).

So we should clean this up by probably removing this method since it's not used anywhere (or exposing the real executable path on the `api.Browser` interface, so it can be read from JS?), making sure the path passed to `LaunchOptions` _is_ used and overrides whatever we actually run, in which case we can avoid doing the runtime lookup, `findExecPath()`, etc.

Reply:
Yeah, I worried about it as well 🤷  `NewAllocator` shouldn't call `findExecPath`, and instead get it from the launch options and only call it if it's empty. Do you mean the same thing?

Reply:
[PW.executablePath](https://playwright.dev/docs/api/class-browsertype#browser-type-launch-option-executable-path)
> Path to a browser executable to run instead of the bundled one. If executablePath is a relative path, then it is resolved relative to the current working directory.

*!!* 

> Note that Playwright only works with the bundled Chromium, Firefox or WebKit, use at your own risk.

Reply:
Done! See: https://github.com/grafana/xk6-browser/pull/246/commits/f1aed4a74fb606f9b921de659689340189e0a24a

`ExecutablePath` returns where the extension expects to see the browser executable. I moved the `findExecutable` directly into that method.